src/commands/auth.rs

```diff
@@ -0,0 +1,26 @@
+use crate::api;
+use crate::credentials;
+use crate::models;
+use api::ApiClient;
+use colored::Colorize;
+use credentials::CredentialsStorage;
+use models::ResultWithDefaultError;
+
+pub struct AuthenticationCommand;
+
+impl AuthenticationCommand {
+    pub async fn execute(
+        api_client: impl ApiClient,
+        credentials_storage: impl CredentialsStorage,
+    ) -> ResultWithDefaultError<()> {
+        let user = api_client.get_user().await?;
+        credentials_storage.persist(user.api_token)?;
+        println!(
+            "{} {}",
+            "Successfully authenticated for user with email:".green(),
+            user.email.green().bold(),
+        );
+
+        return Ok(());
```

Comment:
![image](https://user-images.githubusercontent.com/1716853/122734900-cbb0b580-d27e-11eb-8e45-4dfa857aceb6.png)
clippy doesn't like this

<img width="935" alt="Screenshot 2021-06-21 at 10 52 54" src="https://user-images.githubusercontent.com/1716853/122734943-d8350e00-d27e-11eb-9e2c-8b69b38e1178.png">


You can run

```shell
cargo clippy --fix -Z unstable-options
```

to autofix these, (needs nightly toolchain)